### ISSUE TITLE
The version esbuild@0.8.22  freezes nyc mocha

### ISSUE THREAD

**Issue by deleonio** [evanw/esbuild#595]
Since version 0.8.22 the command `nyc moche` freeze. **Version 0.8.21 works**.

![image](https://user-images.githubusercontent.com/6279703/102041380-a6bf1f80-3dcf-11eb-892f-e5f424ac82ea.png)

To reproduce this as shell script:
```bash
rm -rf demo
mkdir demo
cd demo
npm i -g @leanup/cli@next @leanup/cli-lit-element@next -f
lean create
npm i esbuild@0.8.21 --no-save
npm list --depth=2 | grep esbuild
lean coverage
npm i esbuild@0.8.22 --no-save
npm list --depth=2 | grep esbuild
lean coverage
```

**Comment by evanw** [comment]
Really sorry about the breakage. I'd like to get this fixed ASAP. However, I can't reproduce this. The `lean create` command isn't working for me:

```
 ☀ @leanup/cli-lit-element (v1.1.0-rc.219) execute the following pure command ...
    > npx npm install --loglevel=error --prefer-offline --no-audit --progress=false
 
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: demo-element@1.1.0-rc.219
npm ERR! Found: @leanup/cli@1.1.0-rc.219
npm ERR! node_modules/@leanup/cli
npm ERR!   dev @leanup/cli@"^1.1.0-rc.219" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer @leanup/cli@"~1.1.0-webpack.0" from @leanup/cli-addons@1.1.0-webpack.15
npm ERR! node_modules/@leanup/cli-addons
npm ERR!   dev @leanup/cli-addons@"^1.1.0-rc.217" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See C:\Users\User\AppData\Local\npm-cache\eresolve-report.txt for a full report.
 
npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\User\AppData\Local\npm-cache\_logs\2020-12-14T21_10_52_846Z-debug.log
npm ERR! code 1
npm ERR! path C:\Users\User\Desktop\demo
npm ERR! command failed
npm ERR! command C:\Windows\system32\cmd.exe /d /s /c npm install --loglevel=error --prefer-offline --no-audit --progress=false
 
npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\User\AppData\Local\npm-cache\_logs\2020-12-14T21_10_52_879Z-debug.log
 
 ✘ @leanup/cli command execution ended with error (in 7531 ms)
```

Here are my versions:

```
> node --version
v15.3.0

> npm --version
7.0.14
```

This is on 64-bit Windows 10 in a VM. Let me know if you think there's another way to debug this or if I need a specific version of node and/or npm to be able to run that command.

**Comment by skyrpex** [comment]
It freezes to me as well when transpiling a very simple TS file with `esno`. I'm trying to reproduce it but no luck so far. I'll post something again if I find more info about it.

**Comment by eigilsagafos** [comment]
Could this be the same issue as I was seeing in #596?

**Comment by skyrpex** [comment]
This may be helpful.

esbuild freezes to me if used with `node -r`, for example with `node -r esbuild-register file.ts`.

It freezes on the `startWorkerThreadService.runCallSync` method, the line looks like this: `let status = Atomics.wait(sharedBufferView, 0, 0);`

`file.ts` is just `console.log("test");`

I'm using
```
esbuild-register@1.1.1
esbuild@0.8.22
```

Thing is... if I invoke `esbuild.transformSync` from a simple build.js file, it works as expected and the atomic continues. It looks like this:

```
var esbuild = require("esbuild");

esbuild.transformSync(`console.log("ha");`, {
  sourcefile: `${__dirname}/file.ts`,
  sourcemap: true,
  loader: "ts",
  target: undefined,
  jsxFactory: undefined,
  jsxFragment: undefined,
});
```

**Comment by evanw** [comment]
Thanks for the tip. I can also reproduce it when using `esbuild-register`. The worker is created but the `parentPort` is `null`, so the worker thread is completely disconnected from the main thread. This might be a bug in node itself. Perhaps the `worker_threads` module is unusable as a general-purpose parallelism mechanism. I will push a revert of recent change that makes use of the `worker_threads` library, which should fix this issue.

**Comment by evanw** [comment]
This should now be fixed in version 0.8.23. Again, sorry for the breakage. I now have test coverage for `node -r` so this shouldn't happen again.

**Comment by deleonio** [comment]
Thank you @evanw - and for your advice - sorry.
